### ISSUE TITLE
[Pinot connector] The `IN` condition does not work correctly for string values when using aggregate pushdown

### ISSUE THREAD

**Issue by jpohanka** [trinodb/trino#9133]
**Trino version**: 361

**Description**:

Lets assume that we have a Pinot table `tab` and in this table, we have 3 columns:

- `col_int` - an integer dimensional column
- `col_str` - a string dimensional column
- `quant` - a double column for some quantity

When running an aggregation query with a `WHERE` clause containing an `IN` or `NOT IN` condition on the **integer** column `col_int`, the following queries work as expected:

```
SELECT
  col_int,
  SUM(quant) AS sum_quant
FROM
  pinot.default.tab
WHERE
  col_int IN (value_1, value_2,...)
GROUP BY
  col_in
```

```
SELECT
  col_int,
  SUM(quant) AS sum_quant
FROM
  pinot.default.tab
WHERE
  col_int NOT IN (value_1, value_2,...)
GROUP BY
  col_in
```

However, the situation is **different for the string column** `col_str`. Here, the following aggregate query containing `NOT IN` still works as expected:

```
SELECT
  col_str,
  SUM(quant) AS sum_quant
FROM
  pinot.default.tab
WHERE
  col_str NOT IN (value_1, value_2,...)
GROUP BY
  col_str
```

But when using the `IN` condition  instead of the `NOT IN` condition in the `WHERE` clause, the query returns the result only for the first value `value_1` in the `IN` condition.

**Comment by hashhar** [comment]
cc: @elonazoulay 

@jpohanka Would it be possible for you to try the example queries against this PR https://github.com/trinodb/trino/pull/9098?
It rewrites some of the query generation logic around operators like IN and NOT IN (in addition to other things).

**Comment by jpohanka** [comment]
@hashhar I tested the queries against the PR #9098 , but the issue is still persistent. I have used the source code with the commit [7ebd38e](https://github.com/elonazoulay/presto-1/commit/7ebd38eff49882f1278c3b7c4aaba7eee7fde5f3).

**Comment by elonazoulay** [comment]
Confirmed, adding this test to the fix, will update once the tests pass. Thanks for catching this @jpohanka !

**Comment by jpohanka** [comment]
You're welcome @elonazoulay :) !

**Comment by elonazoulay** [comment]
Looks like the errors are fixed in #9098 - there are some unrelated test failures (memory connector). Let me know if you have any issues.
